test/qdmi/test_driver.cpp

```diff
@@ -0,0 +1,379 @@
+}
+
+TEST_P(DriverTest, QueryDevicePropertyImplemented) {
+  EXPECT_EQ(QDMI_device_query_device_property(device, QDMI_DEVICE_PROPERTY_MAX,
+                                              0, nullptr, nullptr),
+            QDMI_ERROR_INVALIDARGUMENT)
+      << "Devices must implement `QDMI_device_query_device_property`.";
+}
+
+TEST_P(DriverTest, QuerySitePropertyImplemented) {
+  EXPECT_EQ(QDMI_device_query_site_property(
+                device, nullptr, QDMI_SITE_PROPERTY_MAX, 0, nullptr, nullptr),
+            QDMI_ERROR_INVALIDARGUMENT)
+      << "Devices must implement `QDMI_device_query_site_property`.";
+}
+
+TEST_P(DriverTest, QueryOperationPropertyImplemented) {
+  EXPECT_EQ(QDMI_device_query_operation_property(
+                device, nullptr, 0, nullptr, 0, nullptr,
+                QDMI_OPERATION_PROPERTY_MAX, 0, nullptr, nullptr),
+            QDMI_ERROR_INVALIDARGUMENT)
+      << "Devices must implement `QDMI_device_query_operation_property`.";
+}
+
+TEST_P(DriverTest, QueryDeviceVersion) {
+  size_t size = 0;
+  ASSERT_EQ(QDMI_device_query_device_property(
+                device, QDMI_DEVICE_PROPERTY_VERSION, 0, nullptr, &size),
+            QDMI_SUCCESS)
+      << "Devices must provide a version.";
+  std::string value(size - 1, '\0');
+  ASSERT_EQ(QDMI_device_query_device_property(device,
+                                              QDMI_DEVICE_PROPERTY_VERSION,
+                                              size, value.data(), nullptr),
+            QDMI_SUCCESS)
+      << "Devices must provide a version.";
+  EXPECT_FALSE(value.empty()) << "Devices must provide a version.";
+}
+
+TEST_P(DriverTest, QueryDeviceLibraryVersion) {
+  size_t size = 0;
+  ASSERT_EQ(QDMI_device_query_device_property(
+                device, QDMI_DEVICE_PROPERTY_LIBRARYVERSION, 0, nullptr, &size),
+            QDMI_SUCCESS)
+      << "Devices must provide a library version.";
+  std::string value(size - 1, '\0');
+  ASSERT_EQ(QDMI_device_query_device_property(
+                device, QDMI_DEVICE_PROPERTY_LIBRARYVERSION, size, value.data(),
+                nullptr),
+            QDMI_SUCCESS)
+      << "Devices must provide a library version.";
+  ASSERT_FALSE(value.empty()) << "Devices must provide a library version.";
+}
+
+TEST_P(DriverTest, QueryNumQubits) {
+  size_t numQubits = 0;
+  ASSERT_EQ(
+      QDMI_device_query_device_property(device, QDMI_DEVICE_PROPERTY_QUBITSNUM,
+                                        sizeof(size_t), &numQubits, nullptr),
+      QDMI_SUCCESS)
+      << "Devices must provide the number of qubits.";
+  EXPECT_GT(numQubits, 0) << "Number of qubits must be greater than 0.";
+}
+
+TEST_P(DriverTest, QueryDeviceProperties) {
+  EXPECT_EQ(QDMI_device_query_device_property(device, QDMI_DEVICE_PROPERTY_MAX,
+                                              0, nullptr, nullptr),
+            QDMI_ERROR_INVALIDARGUMENT)
+      << "The MAX property is not a valid value for any device.";
+}
```

Comment:
As it stands, this is a redundant test, as the respective call is already tested a few tests above.